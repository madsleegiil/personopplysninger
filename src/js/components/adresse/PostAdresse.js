import React from "react";
import PropTypes from "prop-types";
import Box from "js/components/Box";
import ListElement from "../ListElement";
import { mergeAddress } from "../../utils/text";

const PostAdresse = props => {
  const { adresse1, adresse2, adresse3, postnummer, land } = props;
  const adresse = mergeAddress(adresse1, adresse2, adresse3);
  return (
    <Box header="Postadresse">
      <div className="address-box">
        <ul className="list-column-2">
          {adresse ? (
            <ListElement titleId="adresse.adresse" content={adresse} />
          ) : null}
          {postnummer ? (
            <ListElement titleId="adresse.postnummer" content={postnummer} />
          ) : null}
          {land ? <ListElement titleId="adresse.land" content={land} /> : null}
        </ul>
      </div>
    </Box>
  );
};

PostAdresse.propTypes = {
  adresse1: PropTypes.string,
  adresse2: PropTypes.string,
  adresse3: PropTypes.string,
  land: PropTypes.string,
  postnummer: PropTypes.string
};

PostAdresse.defaultProps = {
  adresse1: "",
  adresse2: "",
  adresse3: "",
  land: "",
  postnummer: ""
};

export default PostAdresse;
