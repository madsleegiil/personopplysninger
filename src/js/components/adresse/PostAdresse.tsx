import React from "react";
import { injectIntl, InjectedIntlProps } from "react-intl";
import Box from "../Box";
import ListElement from "../ListElement";
import { Postadresse } from "../../../types/adresser/postadresse";
import { mergeAddress } from "../../utils/text";

type Props = { postadresse: Postadresse } & InjectedIntlProps;
const PostAdresse = (props: Props) => {
  const { intl } = props;
  const {
    adresse1,
    adresse2,
    adresse3,
    postnummer,
    poststed,
    land
  } = props.postadresse;
  const adresse = mergeAddress(adresse1, adresse2, adresse3);
  return (
    <Box
      header={intl.formatMessage({ id: "adresse.postadresse" })}
      id="postadresse"
    >
      <div className="address-box">
        <ul className="list-column-2">
          {adresse && (
            <ListElement titleId="adresse.adresse" content={adresse} />
          )}
          {postnummer && (
            <ListElement titleId="adresse.postnummer" content={postnummer} />
          )}
          {land && <ListElement titleId="adresse.land" content={land} />}
          {poststed && (
            <ListElement titleId="adresse.poststed" content={poststed} />
          )}
        </ul>
      </div>
    </Box>
  );
};

export default injectIntl(PostAdresse);
