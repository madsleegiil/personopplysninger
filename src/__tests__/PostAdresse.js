import * as React from 'react';
import wrapIntl from 'js/IntlTestHelper';
import PostAdresse from 'js/components/adresse/PostAdresse';
const ReactTestRenderer = require('react-test-renderer');

test('render Postadresse with content', () => {
  const postadresse = {
    adresse1: "12345 HFJEF IQUFJ HFURHGHQBEUQ",
    adresse2: "BATOOM LUNCHAGRAMA,GONGRUPEA,DD",
    adresse3: "GGLAM,10000 HCAMGHU,THAILAND",
    datoFraOgMed: "2018-03-13",
    land: "THA",
    postnummer: "9913"
  };

  const component = ReactTestRenderer.create(wrapIntl(<PostAdresse
    adresse1={postadresse.adresse1}
    adresse2={postadresse.adresse2}
    adresse3={postadresse.adresse3}
    datoFraOgMed={postadresse.datoFraOgMed}
    kilde={postadresse.kilde}
    land={postadresse.land}
    postnummer={postadresse.postnummer}
  />));
  expect(component.toJSON()).toMatchSnapshot();
});
