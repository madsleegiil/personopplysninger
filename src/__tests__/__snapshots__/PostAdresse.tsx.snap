// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Postadresse with content 1`] = `
<div
  className="addresse__box"
>
  <hr
    className="box__linje-bred"
  />
  <div
    className="addresse__tittel"
  >
    <h2
      className="typo-undertittel"
    >
      <span>
        Postadresse
      </span>
    </h2>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      12345 HFJEF IQUFJ HFURHGHQBEUQ
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      BATOOM LUNCHAGRAMA,GONGRUPEA,DD
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      GGLAM,10000 HCAMGHU,THAILAND
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      9913
       
    </p>
  </div>
  <div
    className="adresse__linje"
  >
    <p
      className="typo-normal"
    >
      THA
    </p>
  </div>
  <div
    className="kilde__container"
  >
    <p
      className="typo-undertekst"
    >
      <span>
        Kilde: Folkeregisteret
      </span>
    </p>
    <a
      className="lenke"
      href="https://www.skatteetaten.no/person/folkeregister/"
    >
      <p
        className="typo-normal"
      >
        <span>
          Endre hos Folkeregisteret
        </span>
      </p>
    </a>
  </div>
</div>
`;
